### PR TITLE
Revert civicrm_action_schedule.limit_to boolean column to be NULL

### PR DESCRIPTION
**Pull request by monishdeb**
Overview
----------------------------------------
Revert the boolean field civicrm_action_schedule.limit_to to be NULL and not required, to address issues https://lab.civicrm.org/dev/core/-/issues/3464 and https://lab.civicrm.org/dev/core/-/issues/3465 

In my opinion, limit_to should be int with desired options. 

Comments
----------------------------------------
ping @agileware-justin @eileenmcnaughton @MegaphoneJon @colemanw

### CODE REVIEW

File: CRM/Upgrade/Incremental/php/FiveFortyNine.php [hunk 1]
```diff
@@ -60,6 +60,24 @@ public function upgrade_5_49_beta1($rev): void {
...
+   */
+  public function upgrade_5_49_2($rev) {
+    $this->addtask('Revert civicrm_action_schedule.limit_to to be NULL', 'changeBooleanColumnLimitTo');
+  }
+
+  /**
+   * Revert boolean default civicrm_action_schedule.limit_to to be NULL
+   */
+  public static function changeBooleanColumnLimitTo() {
+    CRM_Core_DAO::executeQuery("ALTER TABLE `civicrm_action_schedule` CHANGE `limit_to` `civicrm_action_schedule` tinyint NULL COMMENT 'Is this the recipient criteria limited to OR in addition to?'", [], TRUE, NULL, FALSE, FALSE);
```
Comment (eileenmcnaughton) [inline CRM/Upgrade/Incremental/php/FiveFortyNine.php:76]: You are actually renaming the field from limit_to to civicrm_action_schedule here

**Comment by eileenmcnaughton** [comment]
Revert civicrm_action_schedule.limit_to to be NULL (CRM_Upgrade_Incremental_php_FiveFortyNine::changeBooleanColumnLimitTo())
Error executing task "Revert civicrm_action_schedule.limit_to to be NULL"

In Error.php line 955:
                           
  [PEAR_Exception]         
  DB Error: no such field

**Comment by agileware-justin** [comment]
Just my 2c. Would prefer to not revert to the previous state, which was also problematic.

Better to forge ahead and change the stored values: 1 (neither), 2 (Limit or Add Recipients), 3 (Also Include) and column to Integer. Fix the core issue with how this was implemented/

**Comment by eileenmcnaughton** [comment]
@monishdeb @agileware-justin we updated the value of this field up update to boolean - is the value we updated it to 'ok' now? I don't see a data update in here

**Comment by eileenmcnaughton** [comment]
@monishdeb to implement @agileware-justin's suggestion I think we would add a pseudoconstant to the xml with a callback in the `SelectValues` class - does that sound right to you? I'm not totally au fait with the back story here so it might be more complicated than that?

**Comment by monishdeb** [comment]
@eileenmcnaughton yeah I agree. Usually, when we find a regression, we tried to revert that code and submit a fix on the master branch as an immediate course of action. In this case, I thought of submitting the revert patch, and then submit a PR that would add three options, updating the RecipientBuilder code and existing unit-tests accordingly. This would be a new necessary fix that would need thorough testing.

**Comment by eileenmcnaughton** [comment]
monishdeb - that makes sense

**Comment by monishdeb** [comment]
I am going to submit a PR against `master` branch, that adds a pseudoconstant to the xml with a callback in the SelectValues class, necessary changes in RecipientBuilder code, updates existing unit-tests.

**Comment by totten** [comment]
Replaced by #23497, which starts from similar commits but adds a few.

**Pull request closed without merging**